src/LiveComponent/src/ComponentWithMultiStepFormTrait.php

```diff
@@ -0,0 +1,283 @@
+            throw new \RuntimeException('No previous forms available.');
+        }
+
+        $this->currentStepName = $previous;
+        $this->getStorage()->persist(\sprintf('%s_current_step_name', self::prefix()), $this->currentStepName);
+
+        $this->form = $this->instantiateForm();
+        $this->formView = null;
+
+        $formData = $this->getStorage()->get(\sprintf(
+            '%s_form_values_%s',
+            self::prefix(),
+            $this->currentStepName,
+        ));
+
+        $this->formValues = $formData;
+        $this->form->setData($formData);
+    }
+
+    #[ExposeInTemplate]
+    public function isFirst(): bool
+    {
+        return $this->currentStepName === $this->stepNames[array_key_first($this->stepNames)];
+    }
+
+    #[ExposeInTemplate]
+    public function isLast(): bool
+    {
+        return $this->currentStepName === $this->stepNames[array_key_last($this->stepNames)];
+    }
+
+    #[LiveAction]
+    public function submit(): void
+    {
+        $this->submitForm();
+
+        if ($this->hasValidationErrors()) {
+            return;
+        }
+
+        $this->getStorage()->persist(
+            \sprintf('%s_form_values_%s', self::prefix(), $this->currentStepName),
+            $this->form->getData(),
+        );
+
+        $this->onSubmit();
+    }
+
+    abstract public function onSubmit();
+
+    /**
+     * @return array<string, mixed>
+     */
+    public function getAllData(): array
+    {
+        $data = [];
+
+        foreach ($this->stepNames as $stepName) {
+            $data[$stepName] = $this->getStorage()->get(\sprintf(
+                '%s_form_values_%s',
+                self::prefix(),
+                $stepName,
+            ));
+        }
+
+        return $data;
+    }
+
+    public function resetForm(): void
+    {
+        foreach ($this->stepNames as $stepName) {
+            $this->getStorage()->remove(\sprintf('%s_form_values_%s', self::prefix(), $stepName));
+        }
+
+        $this->getStorage()->remove(\sprintf('%s_current_step_name', self::prefix()));
+
+        $this->currentStepName = $this->stepNames[array_key_first($this->stepNames)];
+        $this->form = $this->instantiateForm();
+        $this->formView = null;
+        $this->formValues = $this->extractFormValues($this->getFormView());
+    }
+
+    abstract protected function getStorage(): StorageInterface;
+
+    /**
+     * @return class-string<FormInterface>
+     */
+    abstract protected static function formClass(): string;
+
+    abstract protected function getFormFactory(): FormFactoryInterface;
+
+    /**
+     * @internal
+     */
+    protected function instantiateForm(): FormInterface
+    {
+        $options = [];
+
+        if (null !== $this->currentStepName) {
+            $options['current_step_name'] = $this->currentStepName;
+        }
+
+        return $this->getFormFactory()->create(
+            type: static::formClass(),
+            options: $options,
```

Comment:
```suggestion
            static::formClass(),
            $options,
```

no named arguments, as there is no BC promise for this from Symfony side

Reply:
Yes but the second argument is data which will be null